lib/util.js

```diff
@@ -58,6 +59,39 @@ var util = _(require('util')).merge({
     }
     return path.resolve.apply(null, [].slice.call(arguments, 1).concat([givenPath]));
   },
+  
+  extractCoordsFromPath: function (givenPath) {
```

Comment:
Please change this function:

* Move out of `util` into `Editor`, call it `Editor.prototype.extractCoordsFromPath`
* Use the following regular expression and store it on `Editor` itself: `Editor.pathCoordsRegExp = /^(.*?)(\:(\d+))?(\:(\d+))?$/;`
* Return an object that looks like `{path: '/file.txt', coords: {x: 1, y: 1}}`
* You should be able to extract like so: ```
var result = {path: givenPath, coords: {x: 1, y: 1}};
var match = givenPath.match(Slap.pathCoordsRegExp);
if (!match) return result; // should match any string but just in case return default :1:1 coords
result.path = match[1];
if (match[3]) result.coords.y = parseInt(match[3]); // if only one coordinate is given, it's a line number
if (match[5]) result.coords.x = parseInt(match[5]); // if two coordinates are given, the second is a column number
return result;
```

It is OK if the filename itself looks like `whatkindoffilenameisthis:5:8` or `foo:5` or another matching string. The user will simply have to append `:1:1` after the filename in the command, as punishment for poor naming practice. :sunglasses: